c_src/sqlite3_nif.c

```diff
@@ -810,6 +898,13 @@ on_load(ErlNifEnv* env, void** priv, ERL_NIF_TERM info)
     return 0;
 }
 
+static void
+on_unload(ErlNifEnv* caller_env, void* priv_data)
+{
+    assert(env);
+
+    sqlite3_config(SQLITE_CONFIG_MALLOC, NULL);
```

Comment:
```suggestion
    sqlite3_config(SQLITE_CONFIG_MALLOC, &default_alloc_methods);
```

Reply:
It could just be better to not do this at all.

Reply:
Implemented the change.